File: backend/demo/src/main/java/com/example/demo/dto/CourseCreate.java
package com.example.demo.dto;

import com.example.demo.entity.Course;
import com.example.demo.entity.CourseLecture;
import com.example.demo.entity.CourseSection;
import com.example.demo.entity.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CourseCreate {
    User user;
    Course course;
    List<CourseSection> courseSection;
    List<CourseLecture> courseLecture;
    int[] array;
}

File: frontend/angular/src/app/common/course-create.ts
import { Course } from './course';
import { CourseLection } from './course-lection';
import { CourseSection } from './course-section';
import { User } from './user';

export class CourseCreate {
  user: User;
  course: Course;
  courseSection: CourseSection[];
  courseLecture: CourseLection[];
  array: number[];
}
